lib/utils/payment_hash_poller.dart

```diff
@@ -0,0 +1,61 @@
+// ignore_for_file: avoid_print
+import 'dart:async';
+
+import 'package:breez_sdk/bridge_generated.dart';
+import 'package:c_breez/services/injector.dart';
+
+class PaymentHashPoller {
+  final String paymentHash;
+  void Function(bool contains) onDone;
+
+  PaymentHashPoller({
+    required this.onDone,
+    required this.paymentHash,
+  });
+
+  void start() {
+    const int interval = 5;
+    print("Start polling for payment hash: $paymentHash every $interval seconds");
+    final injector = ServiceInjector();
+    final breezLib = injector.breezLib;
+
+    Timer? paymentReceivedTimer;
+
+    paymentReceivedTimer = Timer.periodic(
```

Comment:
paymentReceivedTimer is never stopped.
I think we should have some upper limit on how much time we poll and then cancel the timer.

Reply:
Done 91638d131cfd3fc79ec45e27fe7649f069ec2f21
Cancelled after a minute if it's available.